File: Corporate Flight Bookings.cpp
class Solution {
public:
    vector<int> corpFlightBookings(vector<vector<int>>& bookings, int n) {
        vector<int> res(n, 0);
        vector<int> prefix(n, 0);
        
        for (auto booking : bookings) {
            prefix[booking[0]-1] += booking[2];
            if (booking[1]-1 < n-1) {
                prefix[booking[1]] -= booking[2];
            }
        }
        
        res[0]=prefix[0];
        for (int i=1; i<n; i++) {
            res[i] = res[i-1] + prefix[i];
        }
        
        return res;
    }
};

File: first-phase/week2/corporate-flight-bookings.py
class Solution:
    def corpFlightBookings(self, bookings: List[List[int]], n: int) -> List[int]:
        prefix = [ 0 for _ in range(n+2)]
        for i in bookings:
            prefix[i[0]] += i[2]
            prefix[i[1]+1] -= i[2]
        total = 0
        for j in range(len(prefix)):
            total +=prefix[j]
            prefix[j] = total
        return prefix[1:-1]